e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/T_TwinBrowseOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,249 @@
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.First();
+
+            Assert.NotNull(endpoint.Id);
+            Assert.Equal("ActivatedAndConnected", endpoint.ActivationState);
+            Assert.Equal("Ready", endpoint.EndpointState);
+
+            var nodes = GetAllNodesAsync(endpoint.Id, "Variable").GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            Assert.True(nodes.Count > 150);
+        }
+
+        /// <summary>
+        /// Equivalent to GetSetOfUniqueNodesAsync.
+        /// </summary>
+        private async Task<List<(string NodeId, string NodeClass, bool Children)>> GetNodesAsync(string endpointId, string nodeId = null, string continuationToken = null) {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+            var accessToken = await TestHelper.GetTokenAsync(_context, cts.Token).ConfigureAwait(false);
+            var client = new RestClient(_context.IIoTPlatformConfigHubConfig.BaseUrl) { Timeout = TestConstants.DefaultTimeoutInMilliseconds };
+
+            var request = new RestRequest(Method.GET);
+            request.AddHeader(TestConstants.HttpHeaderNames.Authorization, accessToken);
+
+            if (continuationToken == null) {
+                request.Resource = $"twin/v2/browse/{endpointId}";
+
+                if (!string.IsNullOrEmpty(nodeId)) {
+                    request.AddQueryParameter("nodeId", nodeId);
+                }
+            }
+            else {
+                request.Resource = $"twin/v2/browse/{endpointId}/next";
+                request.AddQueryParameter("continuationToken", continuationToken);
+            }
+
+            var response = await client.ExecuteAsync(request, cts.Token).ConfigureAwait(false);
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            dynamic json = JsonConvert.DeserializeObject(response.Content);
+
+            var result = new List<(string NodeId, string NodeClass, bool Children)>();
+
+            foreach (var node in json.references) {
+                result.Add(
+                    (
+                        node.target?.nodeId?.ToString(),
+                        node.target?.nodeClass?.ToString(),
+                        string.Equals(node.target?.children?.ToString(), "true", StringComparison.OrdinalIgnoreCase)));
+            }
+
+            var responseContinuationToken = json.continuationToken;
+
+            if (responseContinuationToken != null) {
+                result.AddRange(await GetNodesAsync(endpointId, null, responseContinuationToken).ConfigureAwait(false));
+            }
+
+            return result;
+        }
+
+        private async Task<List<(string NodeId, string NodeClass, bool Children)>> GetAllNodesAsync(string endpointId, string nodeClass = null, string nodeId = null) {
```

Comment:
You could also move this methods to TestHelper if they are also usable/useful by other E2E tests

Reply:
Moved to TestHelper